go/vt/sqlparser/sql.y

```diff
@@ -1217,6 +1227,20 @@ index_info:
     $$ = &IndexInfo{Type: string($1), Name: NewColIdent($2), Unique: false}
   }
 
+indexes_or_keys:
+  INDEX
+  {
+    $$ = string($1)
```

Comment:
For rules like this where all matches evaluate to the same thing, you can just declare the evaluation for the first one and omit it for the others. Check out explainable_statement for an example.